Plugins/Encounter/Encounter.cpp

```diff
@@ -200,4 +203,36 @@ ArgumentStack Encounter::SetResetTime(ArgumentStack&& args)
     return Services::Events::Arguments();
 }
 
+ArgumentStack Encounter::GetNumberOfSpawnPoints(ArgumentStack&& args)
+{
+    int32_t retVal = 0;
+
+    if (auto *pEncounter = encounter(args))
+    {
+        retVal = pEncounter->m_nNumSpawnPoints;
+    }
+
+    return Services::Events::Arguments(retVal);
+}
+
+ArgumentStack Encounter::GetSpawnPointByIndex(ArgumentStack&& args)
+{
+    float x = 0.0, y = 0.0, z = 0.0, o = 0.0;
+
+    if (auto *pEncounter = encounter(args))
+    {
+        const auto index = Services::Events::ExtractArgument<int32_t>(args);
```

Comment:
Should add an `ASSERT_OR_THROW(index >= 0);` here.

If you're feeling fancy, add one to `GetEncounterCreatureByIndex` too, which *someone* seems to have forgotten.